### PR TITLE
May 2023 Sprint

### LINKED ISSUE #277: 6. Scholars Discovery endpoints to return docx of grants and pubs for 5 year or 8 year and for individual faculty or whole department
New docx file export (Admin only)

Version 1: 
- Profile Summary with no changes, but the publications from the last 5 years (categorized by the types), instead of 10 most recent - publications
- Active grants within the last 5 years
- No picture and no Scholars@TAMU logo

Version 2: 
- Profile Summary with no changes, but the publications from the last 8 years (categorized by the types), instead of 10 most recent publications.
- No picture and no Scholars@TAMU logo

- [x] One URI for 5 years and another for 8 years
- [x] If faculty VIVO ID used, export the faculty docx file
- [x] If department vivo ID used, export all the department’s faculty docx files (as one file or multiple files)

Example:
- https://api.library.tamu.edu/scholars-discovery/individual/n8ee0b54f/export?type=docx&name=Profile%20Summary

### LINKED ISSUE #310: Organization person totals off by one between UI and zip docx export
Resolved by https://github.com/TAMULib/scholars-discovery/pull/308/commits/4502f641c54ccb6d29f41315a93ab14d103c9368

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/main/java/edu/tamu/scholars/middleware/discovery/model/repo/IndividualRepo.java [hunk 1]
```diff
@@ -52,12 +56,14 @@
...
 import edu.tamu.scholars.middleware.discovery.response.DiscoveryNetwork;
+import edu.tamu.scholars.middleware.utility.DateFormatUtility;
+
 import reactor.core.publisher.Flux;
 
 @Service
 public class IndividualRepo implements IndexDocumentRepo<Individual> {
 
-  private static final Logger logger = LoggerFactory.getLogger(IndividualRepo.class);
+    private static final Logger logger = LoggerFactory.getLogger(IndividualRepo.class);
```
Comment (rmathew1011) [inline src/main/java/edu/tamu/scholars/middleware/discovery/model/repo/IndividualRepo.java:66]: Exrta space here

File: src/main/java/edu/tamu/scholars/middleware/messaging/handler/CustomStompSubProtocolErrorHandler.java [hunk 2]
```diff
@@ -12,9 +12,9 @@ public class CustomStompSubProtocolErrorHandler extends StompSubProtocolErrorHan
 
     @Override
     protected Message<byte[]> handleInternal(StompHeaderAccessor errorHeaderAccessor, byte[] errorPayload, @Nullable Throwable cause, @Nullable StompHeaderAccessor clientHeaderAccessor) {
-    	if (Objects.nonNull(clientHeaderAccessor)) {
-    		errorHeaderAccessor.setDestination(clientHeaderAccessor.getDestination());
-    	}
+        if (Objects.nonNull(clientHeaderAccessor)) {
+            errorHeaderAccessor.setDestination(clientHeaderAccessor.getDestination());
+        }
```
Comment (rmathew1011) [inline src/main/java/edu/tamu/scholars/middleware/messaging/handler/CustomStompSubProtocolErrorHandler.java:17]: This is unusual request - the indentation  on Lines 15 and 17 are highlighted - although only Line 16 has the intended change.

**Review by rmathew1011 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/tamu/scholars/middleware/discovery/model/repo/IndividualRepo.java (hunk 1 above)
Comment (ghost) [reply src/main/java/edu/tamu/scholars/middleware/discovery/model/repo/IndividualRepo.java:66]: image

File: src/main/java/edu/tamu/scholars/middleware/messaging/handler/CustomStompSubProtocolErrorHandler.java (hunk 2 above)
Comment (ghost) [reply src/main/java/edu/tamu/scholars/middleware/messaging/handler/CustomStompSubProtocolErrorHandler.java:17]: image

LGTM

File: src/main/java/edu/tamu/scholars/middleware/discovery/model/repo/IndividualRepo.java (hunk 1 above)
Comment (ghost) [reply src/main/java/edu/tamu/scholars/middleware/discovery/model/repo/IndividualRepo.java:66]: LGTM

File: README.md [hunk 3]
```diff
@@ -54,24 +54,38 @@ Solr is configured via ```spring.data.solr```.
...
 ```bash
-   cd scholars-discovery/solr
+   cd solr
    docker build --tag=scholars/solr .
    docker run -d -p 8983:8983 scholars/solr
 ```
+
+or
+
+Copy `solr/configsets/scholars-discovery` to local Solr installation and run
```
Comment (ghost) [inline README.md]: This alternative development setup provides better performance but requires setup and maintenance locally.

**Review by rmathew1011 — DISMISSED** [review]

File: src/main/resources/defaults/displayViews/defaults/mainContentTemplate.html [hunk 4]
```diff
@@ -9,6 +9,6 @@
   <span class="h3 lead font-weight-normal text-nowrap">
     <span class="ml-1">|</span>
     <span class="ml-1">{{formalize type}}</span>
-    <span class="ml-1"><a href="{{_links.self.href}}" title="View individual in JSON format" download target="_blank"><i class="fa fa-link"><span class="d-none">individual record</span></i></a></span>
+    <span class="ml-1"><a href="{{serviceUrl}}/individual/{{id}}" title="View individual in JSON format" download target="_blank"><i class="fa fa-link"><span class="d-none">individual record</span></i></a></span>
   </span>
 </div>
```
Comment (rmathew1011) [inline src/main/resources/defaults/displayViews/defaults/mainContentTemplate.html]: New line could be added.

File: src/main/resources/defaults/displayViews/processes/mainContentTemplate.html [hunk 5]
```diff
@@ -3,6 +3,6 @@
   <span class="h3 lead font-weight-normal text-nowrap">
     <span class="ml-1">|</span>
     <span class="ml-1">{{formalize type}}</span>
-    <span class="ml-1"><a href="{{_links.self.href}}" title="View individual in JSON format" download target="_blank"><i class="fa fa-link"><span class="d-none">individual record</span></i></a></span>
+    <span class="ml-1"><a href="{{serviceUrl}}/individual/{{id}}" title="View individual in JSON format" download target="_blank"><i class="fa fa-link"><span class="d-none">individual record</span></i></a></span>
   </span>
 </div>
```
Comment (rmathew1011) [inline src/main/resources/defaults/displayViews/processes/mainContentTemplate.html]: Newline could be added.

File: src/main/resources/defaults/displayViews/relationships/mainContentTemplate.html [hunk 6]
```diff
@@ -3,6 +3,6 @@
   <span class="h3 lead font-weight-normal text-nowrap">
     <span class="ml-1">|</span>
     <span class="ml-1">{{formalize type}}</span>
-    <span class="ml-1"><a href="{{_links.self.href}}" title="View individual in JSON format" download target="_blank"><i class="fa fa-link"><span class="d-none">individual record</span></i></a></span>
+    <span class="ml-1"><a href="{{serviceUrl}}/individual/{{id}}" title="View individual in JSON format" download target="_blank"><i class="fa fa-link"><span class="d-none">individual record</span></i></a></span>
   </span>
 </div>
```
Comment (rmathew1011) [inline src/main/resources/defaults/displayViews/relationships/mainContentTemplate.html]: Newline could be added.

File: src/main/resources/defaults/displayViews/defaults/mainContentTemplate.html (hunk 4 above)
Comment (ghost) [reply src/main/resources/defaults/displayViews/defaults/mainContentTemplate.html]: updated

File: src/main/resources/defaults/displayViews/collections/mainContentTemplate.html [hunk 7]
```diff
@@ -22,4 +22,4 @@
     font-weight: 550;
     margin-right: 2px;
   }
-</style>
```
Comment (rmathew1011) [inline src/main/resources/defaults/displayViews/collections/mainContentTemplate.html:25]: Newline could be added.

**Review by rmathew1011 — APPROVED** [review]

**Pull request merged**